File: depend/libcmb/perl/test2.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Cmb;

my $cmb = new Cmb { size_min => 2, size_max => 2 };
my @items = (1..10000);
our $n = 1;
$cmb->cmb_callback($#items+1, \@items, sub { print $n++ . "\n" });

File: depend/libcmb/python/test2.py
#!/usr/bin/env python
from cmb import *

total = 0
choice = 2
num = 1000

items = range(0, num)
count = len(items)

config = CMB(size_min = choice, size_max = choice)

print "Silently enumerating choose-%u from %u:" % (choice, num)
def afunc(config, seq, nitems, items):
    global total
    total += 1
    return 0
cmb_callback(config, count, items, afunc)
print "%u callbacks executed" % total
